### PR TITLE
make `gw_iface` work with net stat

### PR DESCRIPTION
**Pull request by stevenxxiu**
**Descriptions**

Currently `gw_iface` doesn't work with the network speed (unless it's in a Lua function like `${lua rjust ${downspeed ${gw_iface}} 8}`), and doesn't wok not the network graph.

```
conky.text = [[
${gw_iface}\
${downspeed ${gw_iface}}\
${upspeed ${gw_iface}}
${downspeedgraph ${gw_iface} 60,230 7500000}${alignr}\
${upspeedgraph ${gw_iface} 60,230 3000000}
]]
```

This change lets it work with both:

The `downspeed` portion of the code appears to check for this, but it's buggy as it doesn't run `update_gateway_info()`.

Screenshot_2021-05-17_20-08-10

I'm not sure why this broke, as I remember it did work once. This could be related to [Partially revert change from c352069b31b2dbb4d. · brndnmtthws/conky@da2ddb0](https://github.com/brndnmtthws/conky/commit/da2ddb00ed05a772003f4d4ae5fc1156099dc66f), I'm not sure.

### CODE REVIEW

File: src/net_stat.cc [hunk 1]
```diff
@@ -133,9 +132,12 @@ void parse_net_stat_arg(struct text_object *obj, const char *arg,
 
   if (arg == nullptr) { arg = DEFAULTNETDEV; }
 
+#if defined(__linux__)
```
Comment (brndnmtthws) [inline src/net_stat.cc:135]: Move this `#if` so that it _only_ wraps the call to `update_gateway_info()`.

File: src/net_stat.cc [hunk 2]
```diff
@@ -53,7 +53,6 @@
 #if defined(__linux__)
 #include "linux.h"
 #else
-static char e_iface[50] = "empty";
```
Comment (brndnmtthws) [inline src/net_stat.cc:56]: Why is this removed? I still see it referenced below.

**Review by brndnmtthws — CHANGES_REQUESTED** [review]

File: src/net_stat.cc (hunk 2 above)
Comment (stevenxxiu) [reply src/net_stat.cc]: Ah must be because I didn't put the `#if` above in the right place.

File: src/net_stat.cc (hunk 1 above)
Comment (stevenxxiu) [reply src/net_stat.cc]: Fixed now.

**Comment by stevenxxiu** [comment]
@brndnmtthws Mind looking at this again?
